macros/cast_timestamp.sql

```diff
@@ -16,9 +16,9 @@
 
 {%- if target.type ==  'postgres' -%} 
     {%- if cast_as in ('timestamp_ntz','timestamp') -%}
-        {{val}}::TIMESTAMP
+        {{val}}::TIMESTAMP WITH TIME ZONE AT TIME ZONE 'UTC'
     {%- elif cast_as == 'timestamp_tz' -%}
-        {{val}}::TIMESTAMP AT TIME ZONE 'UTC'
+        {{val}}::TIMESTAMP WITH TIME ZONE
```

Comment:
I'm confused by the changes here. It looks like previously if 'timestamp_tz' then the timestamp is being cast to 'UTC', but now if 'timestamp_ntz' or 'timestamp' then the timestamp is being cast to 'UTC'.

Reply:
I've rethink my thoughts about it and rolled back it. But I'm still a bit unsure about  `TIMESTAMP AT TIME ZONE 'UTC'` and `{{val}}::TIMESTAMP WITH TIME ZONE`. According to test Postgres SQL query:

```
SELECT 
collector_tstamp
, pg_typeof(collector_tstamp)
, collector_tstamp::varchar::TIMESTAMP WITH TIME ZONE
, collector_tstamp::varchar::TIMESTAMP AT TIME ZONE 'UTC'
FROM 
(select '2018-01-15 12:05:44.992+02' as collector_tstamp) as a
```
we see:
![image](https://user-images.githubusercontent.com/100302298/182616412-6559859b-30b1-41e4-aca9-7f5160089bff.png)
Which one is correct for us?

Reply:
I'm honestly not sure. It depends on what the original intent of this macro was, which I don't really understand. Maybe compare the outputs to the snowflake outputs? That way even if we don't totally understand what this macro is supposed to be doing we at least know it does the same thing for postgres that it does for snowflake. In the process of doing that we may understand what's going on.

Reply:
Let's left it untouched, because current logic across dbs is consistent:
```
SELECT 
collector_tstamp
, typeof(collector_tstamp::VARIANT)
, TO_TIMESTAMP_TZ(collector_tstamp) AS res_tz
, TO_TIMESTAMP_TZ(TO_TIMESTAMP_NTZ(collector_tstamp)::VARCHAR || '+00', 'YYYY-MM-DD HH24:MI:SS.FFTZH') AS res
FROM 
(select '2018-01-15 12:05:44.992+02' as collector_tstamp) as a
```
![Screenshot 2022-08-04 at 14 16 39](https://user-images.githubusercontent.com/100302298/182834429-d024d119-ff22-4630-abef-af5e7a3ac3ae.png)

So comparing with previous screen we can see that
`TO_TIMESTAMP_TZ(TO_TIMESTAMP_NTZ(collector_tstamp)::VARCHAR || '+00', 'YYYY-MM-DD HH24:MI:SS.FFTZH') <=> collector_tstamp::varchar::TIMESTAMP AT TIME ZONE 'UTC'`

Sorry for disturbing with this.